File: AdTownsBackend.net/AdTowns/Models/Registration.cs
namespace AdTowns.Models;

public class Registration
{
    public Guid Id { get; set; }
    public string ReferenceId { get; set; } = null!;
    public string Name { get; set; } = null!;
    public string? Business { get; set; }
    public string Phone { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string City { get; set; } = null!;
    public RegistrationType Type { get; set; }
    public string? Message { get; set; }
    public bool TermsAccepted { get; set; }
    public string? IpAddress { get; set; }
    public string? UserAgent { get; set; }
    public DateTime SubmittedAt { get; set; }
    public bool EmailSentToAdmin { get; set; }
    public bool EmailSentToUser { get; set; }
    public DateTime? EmailSentAt { get; set; }
}

File: react-landing/src/api/registration.ts
import { request } from './http';

// Types for Registration API

export interface RegistrationPayload {
  name: string;
  business?: string;
  phone: string;
  email: string;
  city: string;
  type: 'vendor' | 'buyer' | 'referrer';
  message?: string;
  termsAccepted: boolean;
}

export interface RegistrationSuccessResponse {
  success: true;
  message: string;
  data: {
    referenceId: string;
    name: string;
    email: string;
    city: string;
    type: string;
    submittedAt: string;
  };
}

export interface ValidationErrorResponse {
  success: false;
  code: 'VALIDATION_ERROR';
  message: string;
  errors: {
    [key: string]: string;
  };
}

export interface DuplicateEntryResponse {
  success: false;
  code: 'DUPLICATE_ENTRY';
  message: string;
  field: string;
}

export type RegistrationResponse = 
  | RegistrationSuccessResponse 
  | ValidationErrorResponse 
  | DuplicateEntryResponse;

/**
 * Register a new interest
 * @param payload - The registration data
 * @returns The server response
 */
export async function register(payload: RegistrationPayload): Promise<RegistrationResponse> {
  return request<RegistrationResponse>('/api/register', {
    method: 'POST',
    body: JSON.stringify(payload),
  });
}

